connectors/byoei.py

```diff
@@ -370,11 +356,40 @@ async def prepare_index(
             else:
                 logger.debug("Index %s already has mappings. Skipping...", index)
             return
+        else:
+            raise IndexMissing(f"Index {index} does not exist!")
+
+    async def _override_index(self, index, docs=None, mappings=None, settings=None):
+        """Override the index with new mappings and settings.
+
+        If the index with such name exists, it's deleted and then created again
+        with provided mappings and settings. Otherwise index is just created.
+
+        After that, provided docs are inserted into the index.
+
+        This method is supposed to be used only for testing - framework is not
+        supposed to create/delete indices at all, Kibana is responsible for
+        this logic.
+        """
+
+        expand_wildcards = ElasticServer._get_expand_wildcards(index)
+
+        exists = await self.client.indices.exists(
+            index=index, expand_wildcards=expand_wildcards
+        )
+
+        if exists:
+            logger.debug(f"{index} exists, deleting...")
```

Comment:
I would use `warning`

Reply:
I used the logger call from the original method and `debug` might make sense - this method is supposed to be called only from `kibana.py` thus Kibana somehow expects this and does not really care much IMO. I can change to warning too, just not sure if it's needed